skrifa/src/outline/mod.rs

```diff
@@ -0,0 +1,448 @@
+    Cff,
+    /// PostScript outlines sourced from the `CFF2` table.
+    Cff2,
+}
+
+/// Specifies the hinting strategy.
+#[derive(Copy, Clone, PartialEq, Eq, Default, Debug)]
+pub enum Hinting {
+    /// Hinting is disabled.
+    #[default]
+    None,
+    /// Will apply native embedded hints.
+    Native,
+}
+
+/// Information and adjusted metrics generated while drawing an outline glyph.
+#[derive(Copy, Clone, Default, Debug)]
+pub struct AdjustedMetrics {
+    /// True if the underlying glyph contains flags indicating the
+    /// presence of overlapping contours or components.
+    pub has_overlaps: bool,
+    /// If present, an adjusted left side bearing value generated by the
+    /// scaler.
+    pub lsb: Option<f32>,
+    /// If present, an adjusted advance width value generated by the
+    /// scaler.
+    pub advance_width: Option<f32>,
+}
+
+/// Options that define how a glyph is drawn to a pen.
+pub struct DrawSettings<'a> {
+    instance: DrawInstance<'a>,
+    memory: Option<&'a mut [u8]>,
+}
+
+impl<'a> DrawSettings<'a> {
+    /// Creates settings for an unhinted draw operation with the given size and
+    /// location in variation space.
+    pub fn new(size: Size, location: impl Into<LocationRef<'a>>) -> Self {
+        Self {
+            instance: DrawInstance::Unhinted(size, location.into()),
+            memory: None,
+        }
+    }
+
+    /// Creates settings for a native hinted draw operation.
+    ///
+    /// The font size, location in variation space and hinting mode are
+    /// defined by the current configuration of the given hinting instance.
+    pub fn native_hinted(instance: &'a NativeHintingInstance) -> Self {
+        Self {
+            instance: DrawInstance::NativeHinted(instance),
+            memory: None,
+        }
+    }
+
+    /// Builder method to associate a user memory buffer to be used for
+    /// temporary allocations during drawing.
+    ///
+    /// The required size of this buffer can be computed using the
+    /// [`OutlineGlyph::draw_memory_size`] method.
+    ///
+    /// If not provided, any necessary memory will be allocated internally.
+    pub fn with_memory(mut self, memory: Option<&'a mut [u8]>) -> Self {
+        self.memory = memory;
+        self
+    }
+}
+
+enum DrawInstance<'a> {
+    Unhinted(Size, LocationRef<'a>),
+    NativeHinted(&'a NativeHintingInstance),
+}
+
+impl<'a, L> From<(Size, L)> for DrawSettings<'a>
+where
+    L: Into<LocationRef<'a>>,
+{
+    fn from(value: (Size, L)) -> Self {
+        DrawSettings::new(value.0, value.1.into())
+    }
+}
+
+impl From<Size> for DrawSettings<'_> {
+    fn from(value: Size) -> Self {
+        DrawSettings::new(value, LocationRef::default())
+    }
+}
+
+impl<'a> From<&'a NativeHintingInstance> for DrawSettings<'a> {
+    fn from(value: &'a NativeHintingInstance) -> Self {
+        DrawSettings::native_hinted(value)
+    }
+}
+
+/// A scalable glyph outline.
+#[derive(Clone)]
```

Comment:
Debug?

Reply:
This contains a lot of deeply nested read-fonts types that require traversal for Debug :( Happy to do a custom impl that shows something, but I'd like to defer that.